kzg/src/eip_7594.rs

```diff
@@ -414,69 +414,40 @@ pub fn recover_cells_and_kzg_proofs<
 >(
     recovered_cells: &mut [[TFr; FIELD_ELEMENTS_PER_CELL]],
     recovered_proofs: Option<&mut [TG1]>,
-    cell_indicies: &[usize],
+    cell_indices: &[usize],
     cells: &[[TFr; FIELD_ELEMENTS_PER_CELL]],
     s: &TKZGSettings,
 ) -> Result<(), String> {
-    if recovered_cells.len() != CELLS_PER_EXT_BLOB
-        || recovered_proofs
-            .as_ref()
-            .is_some_and(|it| it.len() != CELLS_PER_EXT_BLOB)
-    {
-        return Err("Invalid output array length".to_string());
-    }
+    let cells_len = cells.len();
+    let proofs_len = recovered_proofs.as_ref().map_or(0, |it| it.len());
 
-    if cells.len() != cell_indicies.len() {
-        return Err(
-            "Cell indicies mismatch - cells length must be equal to cell indicies length"
-                .to_string(),
-        );
+    if !(recovered_cells.len() == CELLS_PER_EXT_BLOB && 
+          (proofs_len == 0 || proofs_len == CELLS_PER_EXT_BLOB) && 
+          cells_len == cell_indices.len() && 
+          cells_len <= CELLS_PER_EXT_BLOB && 
+          cells_len >= CELLS_PER_EXT_BLOB / 2) {
+        return Err("Cell indices mismatch - invalid length".into());
     }
 
-    if cells.len() > CELLS_PER_EXT_BLOB {
-        return Err("Cell length cannot be larger than CELLS_PER_EXT_BLOB".to_string());
-    }
-
-    if cells.len() < CELLS_PER_EXT_BLOB / 2 {
-        return Err(
-            "Impossible to recover - cells length cannot be less than CELLS_PER_EXT_BLOB / 2"
-                .to_string(),
-        );
-    }
-
-    for cell_index in cell_indicies {
-        if *cell_index >= CELLS_PER_EXT_BLOB {
-            return Err("Cell index cannot be larger than CELLS_PER_EXT_BLOB".to_string());
-        }
+    if cell_indices.iter().any(|&index| index >= CELLS_PER_EXT_BLOB) {
+        return Err("Cell index cannot be larger than CELLS_PER_EXT_BLOB".into());
     }
 
+    let null_cell = TFr::null();
     for cell in recovered_cells.iter_mut() {
-        for fr in cell {
-            *fr = TFr::null();
-        }
+        cell.fill(null_cell);
     }
 
-    for i in 0..cells.len() {
-        let index = cell_indicies[i];
-
-        for j in 0..FIELD_ELEMENTS_PER_CELL {
-            if !recovered_cells[index][j].is_null() {
-                return Err("Invalid output cell".to_string());
-            }
-        }
-
-        recovered_cells[index] = cells[i].clone();
+    for (i, &index) in cell_indices.iter().enumerate() {
+        recovered_cells[index].copy_from_slice(&cells[i]);
     }
```

Comment:
You've removed `is_null` check for cell - this will now not throw error if `cell_indices` array has duplicate indices. I believe, some tests on CI will fail because of this check.